crates/dpapi/src/crypto/mod.rs

```diff
@@ -174,35 +179,38 @@ pub fn content_encrypt(
 }
 
 pub fn kdf(algorithm: HashAlg, secret: &[u8], label: &[u8], context: &[u8], length: usize) -> CryptoResult<Vec<u8>> {
-    use rust_kbkdf::{kbkdf, CounterMode, InputType, KDFMode, SpecifiedInput};
-
-    let mut derived_key = vec![0; length];
-
-    macro_rules! kdf {
-        ($sha:ident) => {{
-            let key = HmacShaPrfKey::new(secret);
-            let mut hmac = $sha::new();
-
-            // KDF(HashAlg, KI, Label, Context, L)
-            // where KDF is SP800-108 in counter mode.
-            kbkdf(
-                &KDFMode::CounterMode(CounterMode { counter_length: 32 }),
-                &InputType::SpecifiedInput(SpecifiedInput { label, context }),
-                &key,
-                &mut hmac,
-                &mut derived_key,
-            )?;
-        }};
+    use hmac_pre::Hmac;
+    use kbkdf::{Counter, Kbkdf, Params};
+    use kout::*;
+
+    macro_rules! derive_key {
+        ($prf_ty:ty, { $($l_value:expr => $l_ty:ty,)* }) => {
+            $(
+                if $l_value == length {
+                    return Ok(Counter::<$prf_ty, $l_ty>::default()
+                        .derive(Params::builder(secret).with_label(label).with_context(context).use_counter(true).use_l(true).build())?
+                        .to_vec());
+                }
+            )*
+        };
     }
```

Comment:
Key length is a type parameter in the `kbkdf` implementation, but our key length is a runtime value. so, we need to move from runtime value to compile time.
The only way to solve this is to check a variable value in runtime and specify the needed type parameter. I wrote this macro to simplify this process (see usage below)